src/HttpRequestAdapter.php

```diff
@@ -77,7 +77,7 @@ public function getUri($path)
         $vars = $this->getServerVars();
 
         $uri = '';
-        $protocol = (isset($vars['HTTPS']) || $vars['SERVER_PORT'] === '443') ? 'https' : 'http';
+        $protocol = (isset($vars['HTTPS']) || isset($vars['SERVER_PORT']) && $vars['SERVER_PORT'] === '443') ? 'https' : 'http';
```

Comment:
Could you please put the && conditions in brackets to avoid ambiguity of the precedence with the first isset check?

Reply:
Done.